third-party/realdds/src/dds-topic-reader-thread.cpp

```diff
@@ -39,7 +45,44 @@ void dds_topic_reader_thread::run( qos const & rqos )
     status_mask << eprosima::fastdds::dds::StatusMask::subscription_matched();
     //status_mask << eprosima::fastdds::dds::StatusMask::data_available();
     _reader = DDS_API_CALL( _subscriber->get()->create_datareader( _topic->get(), rqos, this, status_mask ) );
-    _th.start();
+    
+    _th = std::thread(
+        [this, name = _topic->get()->get_name()]()
+        {
+            eprosima::fastdds::dds::WaitSet wait_set;
+            auto & condition = _reader->get_statuscondition();
+            condition.set_enabled_statuses( eprosima::fastdds::dds::StatusMask::data_available() );
+            wait_set.attach_condition( condition );
+
+            wait_set.attach_condition( _stopped );
+
+            while( ! _stopped.get_trigger_value() )
+            {
+                _stopped.set_trigger_value( false );
+
+                eprosima::fastdds::dds::ConditionSeq active_conditions;
+                wait_set.wait( active_conditions, eprosima::fastrtps::c_TimeInfinite );
+
+                if( _stopped.get_trigger_value() )
+                    break;
+
+                rsutils::time::stopwatch stopwatch;
+                _on_data_available();
+                if( stopwatch.get_elapsed() > std::chrono::milliseconds( 500 ) )
```

Comment:
Why 500? Should be configurable, maybe FPS dependent.
Not for this PR but we should add a task to do it later

Reply:
Right, can be whatever we deem is right, I picked half a second rather arbitrarily.

Reply:
Added to the [todo list](https://rsjira.intel.com/browse/LRS-585)